### PR TITLE
Add transforms from nunit-summary project

### LINKED ISSUE #7: Add transforms from nunit-summary project
In nunit/nunit-summary#19 we decided to move the transforms from the nunit-summary project into this project.

### PR DESCRIPTION
**Pull request by CharliePoole**
**Do Not Merge Yet**
Fixes #7 

I don't want to make any code changes as a part of this issue, but let's decide

1. If we want all eight of the transforms in this project - i.e. are they all useful?
2. How we want to structure them - I currently have them split into two folders for v2 and v3.
3. What other files (readme, samples) we want here

### CODE REVIEW

**Comment by ChrisMaddock** [comment]
I don't see the harm in keeping them all here - and letting people access what they want. The structure looks good to me.

It would be nice to have a readme per subdir, like the current junit transform has. I think it would also be useful to be able to visualise each transform - which is possibly just a link to this page? http://nunit.org/nunit-summary/ I don't think it's worth putting much more effort into this above that, until users suggest otherwise. 🙂 

The main README.md will want updating to include these, if that wasn't already on your list.

**Comment by CharliePoole** [comment]
@ChrisMaddock I took your suggestions. I did use a link to the examples I created a while back, although those should eventually be replaced. They use static files and will be out of date if any changes are made to the transforms.

File: nunit2-summary/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# NUnit 3 Summary Transforms -- Charlie Poole
```
Comment (ChrisMaddock) [inline nunit2-summary/README.md]: This should be NUnit 2.

File: nunit2-summary/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+* `html-report-v2.xslt` creates a report similar to what the console itself displays in html format.
+* `html-summary-v2.xslt` creates the summary report alone in html format.
+* `text-report-v2.xslt` creates a report similar to what the console itself displays in text format.
+* `text-summary-v2.xslt` creates the summary report alone in text format.
+
+See our [website](http://nunit.org/nunit-summary) for samples of the report output.
+
+All the transforms require an input XML file in NUnit V2 format. To apply the transform, you need to use a program that can apply an XSLT transform to an XML file.
+
+> Reviewers: I'd like to offer suggestions here. Ideas? Alternatively, we could supply a program (I have one).
```
Comment (ChrisMaddock) [inline nunit2-summary/README.md]: I personally think 'how to use an xslt' is out of scope - I'd assume user's with advanced enough knowledge to browsing xslt transforms know what they're looking at.

At most, it may be worth a mention that xslt's can be passed into the console - that's NUnit specific knowledge which may not be common place.

File: nunit3-summary/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Naturally, if you use one of the HTML transforms, you will want to change the file type of the result output.
+
+Note that the `--result` option may be repeated to create several reports. If you use the above command-line, the default `TestResult.xml` will not be saved. If you want it you should use a command like this:
+
+```
+nunit3-console.exe my.test.dll --result=my.test.summary.txt;transform=text-summary.xslt --result=TestResult.xml
+```
+
+If you want to use one of the transforms separately, after the test run, you will need to use a program that can apply an XSTL transform to an XML file. Note that the input file must be in NUnit3 format.
```
Comment (ChrisMaddock) [inline nunit3-summary/README.md]: XSTL -> XSLT

File: nunit3-summary/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+> Reviewers: I'd like to offer suggestions here. Ideas? Alternatively, we could supply a program (I have one) or add an option to nunit3 console that lets it do it.
```
Comment (ChrisMaddock) [inline nunit3-summary/README.md]: As above - I think this is out of scope. I think the sentence above is sufficient.

File: nunit2-summary/README.md (hunk 2 above)
Comment (ChrisMaddock) [inline nunit2-summary/README.md]: Ah - I see you mentioned the console on the NUnit 3 versions. Fair enough. (Was this a new feature to NUnit 3? I had limited experience with v2, we were pinned on 2.5.7.)

**Comment by ChrisMaddock** [comment]
And yes - I agree the link could become out of date - but hopefully that's a concern of any future maintainer. 😄

File: nunit2-summary/README.md (hunk 2 above)
Comment (CharliePoole) [inline nunit2-summary/README.md]: Yes, new in 3.0.
